src/interface/fsparse.jl

```diff
@@ -130,6 +130,38 @@ end
 
 fsprandn(args...) = fsprand(args..., randn)
 
+## randsubseq & randsubseq! copied from Random/src/misc.jl
+
+randsubseq(r::AbstractRNG, A::AbstractArray{T}, p::Real) where {T} =
+    randsubseq!(r, T[], A, p)
+
+# Fill S (resized as needed) with a random subsequence of A, where
+# each element of A is included in S with independent probability p.
+function randsubseq!(r::AbstractRNG, S::AbstractArray, A::AbstractArray, p::Real)
+    Base.require_one_based_indexing(S, A)
+    0 <= p <= 1 || throw(ArgumentError("probability $p not in [0,1]"))
+    n = prod(size(A), init=Int64(1))  # length
+    p == 1 && return copyto!(resize!(S, n), A)
```

Comment:
this line is fine to use prod(size(A)) because we know it won't overflow because we are about to materialize it.